fierce/fierce.py

```diff
@@ -359,7 +360,11 @@ def fierce(**kwargs):
         if record is None or record.rrset is None:
             continue
 
-        ip = ipaddress.IPv4Address(record[0].address)
+        ips = [rr.address for rr in record.rrset]
+        if len(wildcard_ips.intersection(ips)) != 0:
```

Comment:
We can simplify this to `if wildcard_ips.intersection(ips)`.

This also brings up an interesting question. Should we `continue` if _any_ IP matches a wildcard IP (i.e. what we're purposing here), or should we attempt to process IPs that _don't_ match the wildcard? In other words, process IPs here that don't match the wildcard rather than discarding them all.

Reply:
I was assuming that it's extremely unlikely for a wildcard subdomain and a "something else" subdomain to have any overlap (unless they're identical) but I suppose it's possible It's a big world out there.

But by the same token, any two wildcard subdomains should have identical `A` sets, so the conservative approach would be to do a set equality check rather than an intersection test. If my assumption holds true, the behavior is no different; if it doesn't hold true, then... well, Python is very accessible and easy to change for a one-off run. :-)

As it stands, we're only looking at the first IP address in the list, but if that stops being true, then I agree that filtering the list to remove any wildcard-subdomain matches would be the best answer.

Reply:
OK, I've convinced myself, requiring a match on the entire wildcard list seems better, and it can always be loosened later.

It's hard to think of sites to query as a reality check, but `slack.com` is one example.

Reply:
(`github.io` has multiple wildcard IPs, but I don't think it has any fixed subdomains.)

Reply:
I agree! It's probably best to start with code _more similar_ to the current code, then work toward looser restrictions later if necessary.